.github/workflows/bench.yml

```diff
@@ -0,0 +1,86 @@
+name: BENCHMARK
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: testnet
+
+jobs:
+  hardware-benchmarks:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          echo "HOME=/home/actions" >> $GITHUB_ENV
+          pallet=$(grep "impl pallet_" runtime/src/lib.rs | grep "Config for Runtime" | sed 's/impl pallet_//' | cut -f1 -d: | sort)
+          for i in $pallet; do
+            mkdir -p pallets/$i/src
+          done
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2022-03-21
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        continue-on-error: true
+
+      - name: Balances
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: balances
+        run: |
+          mkdir -p pallets/$PALLET/src
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Difficulty
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: difficulty
+        run: |
+          mkdir -p pallets/$PALLET/src
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Rewards
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: rewards
+        run: |
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Timestamp
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: timestamp
+        run: |
+          mkdir -p pallets/$PALLET/src
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Creditcoin
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: creditcoin
+        run: |
+          OUTPUT=./pallets/$PALLET/src/weights.rs
+          ./target/release/creditcoin-node benchmark --chain dev --steps=16 --repeat=10 --pallet pallet_$PALLET --extrinsic='*' --execution wasm --wasm-execution=compiled --heap-pages=10000 --output $OUTPUT
+          sed -i s/pallet_$PALLET/super/ $OUTPUT
```

Comment:
I am worried that we may miss some pallets in the future b/c this list is maintained by hand. Can we have some sort of a check that will ensure no new pallets have been added to the node which we have to list here ?

Reply:
Since we are not considering external pallets, we might have to introduce new ci steps for every new pallets to be considered.

Reply:
Given that we already have differences in how weights are generated using a matrix job may not be the best choice. 

@CAGS295 are you happy with the weights for the creditcoin pallet being generated using less steps than the others? If yes we may consider this item as resolved.

Reply:
The pipeline is using one job with multiple steps. Only the first step compiles, others inherits.